CMakeLists.txt

```diff
@@ -161,34 +194,32 @@ if(APPLE)
   # !!! FIXME: check OUTPUT_NAME value.
   set_target_properties(SDL2 PROPERTIES COMPILE_DEFINITIONS "_THREAD_SAFE")
   target_link_libraries(SDL2 PRIVATE
-    "-Wl,-compatibility_version,${DYLIB_COMPAT_VERSION}"
-    "-Wl,-current_version,${DYLIB_CURRENT_VERSION}"
+    "-Wl,-compatibility_version,${SDL_DYLIB_COMPAT_VERSION_MICRO}"
+    "-Wl,-current_version,${SDL_DYLIB_CURRENT_VERSION_MICRO}"
     "-Wl,-framework,AppKit"
   )
   set_target_properties(SDL2 PROPERTIES
-      OUTPUT_NAME "SDL2-2.0.0"
+      OUTPUT_NAME "SDL2-2.0"
+      SOVERSION "0"
   )
 elseif(UNIX AND NOT ANDROID)
   # !!! FIXME: check OUTPUT_NAME value
   set_target_properties(SDL2 PROPERTIES
-    VERSION "${PROJECT_VERSION}"
-    SOVERSION "0"
     OUTPUT_NAME "SDL2-2.0"
+    VERSION "${SDL_SO_VERSION}"
+    SOVERSION "${SDL_SO_VERSION_MAJOR}"
   )
 elseif(WIN32)
   set_target_properties(SDL2 PROPERTIES DEFINE_SYMBOL "DLL_EXPORT")
   # avoid DLL having 'lib' prefix with MinGW
   set_target_properties(SDL2 PROPERTIES
-    VERSION "${PROJECT_VERSION}"
-    SOVERSION "0"
     OUTPUT_NAME "SDL2"
     PREFIX ""
   )
 elseif(OS2)
   set_target_properties(SDL2 PROPERTIES
-    VERSION "${PROJECT_VERSION}"
-    SOVERSION "0"
     OUTPUT_NAME "SDL2"
+    VERSION "${SDL_VERSION}"
```

Comment:
_Should_ not be needed. (Also, since os/2 support is gone in sdl3, this part can go too.)